packages/gatsby/src/services/types.ts

```diff
@@ -0,0 +1,41 @@
+import { WebsocketManager } from "../utils/websocket-manager"
+import { AnyEventObject, Actor } from "xstate"
+import { Compiler } from "webpack"
+import { Span } from "opentracing"
+import { Express } from "express"
+import JestWorker from "jest-worker"
+import { IProgram } from "../commands/types"
+import { Store } from "../.."
+import { GraphQLRunner } from "../query/graphql-runner"
+
+export interface IGroupedQueryIds {
+  pageQueryIds: string[]
+  staticQueryIds: string[]
+}
+
+export interface IMutationAction {
+  type: string
+  payload: unknown[]
+}
+export interface IBuildContext {
```

Comment:
I'm not sure but I feel like a lot of these typings aren't used

Reply:
They're used in other services that will be in future PRs. I thought it simpler to add the compete context type, as otherwise it could get quite confusing to merge.

Reply:
I find it harder because now I don't know if they eventually be used

Reply:
Trust me, they all will be used 😉 . But to be serious, I get your point. I'll see if I can separate them in a way that doesn't cause merge conflicts

Reply:
Done